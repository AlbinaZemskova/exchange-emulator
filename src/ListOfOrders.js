import React, { Fragment } from 'react';
import {
  Row,
  Col,
} from 'reactstrap';
import PropTypes from 'prop-types';


function ListOfOrders({ listOfOrders }) {
  return (
    <Fragment>
      {listOfOrders.map((index, i) => (
        <Row key={i} className="h5">
          <Col lg="5">
            $
            {index.price}
          </Col>
          <Col lg="2">
            x
          </Col>
          <Col lg="5">
            {index.count}
          </Col>
        </Row>
      ))}
    </Fragment>
  );
}

ListOfOrders.propTypes = {
  listOfOrders: PropTypes.arrayOf(
    PropTypes.exact({
      price: PropTypes.number.isRequired,
      count: PropTypes.number.isRequired,
    }).isRequired,
  ).isRequired,
};

export default ListOfOrders;
