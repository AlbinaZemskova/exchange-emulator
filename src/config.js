/* eslint-disable import/prefer-default-export */
export const config = {
  buyOrders: [
    {
      price: 9900,
      count: 30,
    },
    {
      price: 400,
      count: 30,
    },
    {
      price: 300,
      count: 30,
    },
    {
      price: 200,
      count: 30,
    },
    {
      price: 4000,
      count: 30,
    },
    {
      price: 5200,
      count: 40,
    },
    {
      price: 5700,
      count: 300,
    },
    {
      price: 6000,
      count: 100,
    },
    {
      price: 7000,
      count: 200,
    },
    {
      price: 8000,
      count: 300,
    },
    {
      price: 8700,
      count: 310,
    },
    {
      price: 9900,
      count: 100,
    },
    {
      price: 9000,
      count: 350,
    },
    {
      price: 10000,
      count: 420,
    },
    {
      price: 11500,
      count: 440,
    },
  ],
  sellOrders: [
    {
      price: 7000,
      count: 40,
    },
    {
      price: 8000,
      count: 100,
    },
    {
      price: 5000,
      count: 100,
    },
    {
      price: 11000,
      count: 400,
    },
    {
      price: 22000,
      count: 602,
    },
    {
      price: 8000,
      count: 100,
    },
    {
      price: 9900,
      count: 590,
    },
    {
      price: 10000,
      count: 600,
    },
    {
      price: 9900,
      count: 400,
    },
    {
      price: 11400,
      count: 630,
    },
    {
      price: 12000,
      count: 670,
    },
    {
      price: 10000,
      count: 630,
    },
    {
      price: 50000,
      count: 670,
    },
    {
      price: 22400,
      count: 550,
    },
    {
      price: 23500,
      count: 700,
    },
  ],

  history: [],
};
