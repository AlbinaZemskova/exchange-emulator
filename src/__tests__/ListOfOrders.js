import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import ListOfOrders from '../ListOfOrders';
import { config } from '../config';
import 'jest-prop-type-error';

Enzyme.configure({ adapter: new Adapter() });

describe('<ListOfOrders />', () => {
  let wrapper;

  describe('render', () => {
    it('render orders for buy', () => {
      wrapper = shallow(
        <ListOfOrders listOfOrders={config.buyOrders} />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('render orders for sell', () => {
      wrapper = shallow(
        <ListOfOrders listOfOrders={config.sellOrders} />,
      );
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('checking correct proptypes', () => {
      expect(() => (
        <ListOfOrders listOfOrders={{}} />
      )).toThrow();
    });
  });
});
