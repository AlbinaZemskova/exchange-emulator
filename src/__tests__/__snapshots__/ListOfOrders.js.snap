// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListOfOrders /> render render orders for buy 1`] = `
<Fragment>
  <Row
    className="h5"
    key="0"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      9900
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      30
    </Col>
  </Row>
  <Row
    className="h5"
    key="1"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      400
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      30
    </Col>
  </Row>
  <Row
    className="h5"
    key="2"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      300
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      30
    </Col>
  </Row>
  <Row
    className="h5"
    key="3"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      200
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      30
    </Col>
  </Row>
  <Row
    className="h5"
    key="4"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      4000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      30
    </Col>
  </Row>
  <Row
    className="h5"
    key="5"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      5200
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      40
    </Col>
  </Row>
  <Row
    className="h5"
    key="6"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      5700
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      300
    </Col>
  </Row>
  <Row
    className="h5"
    key="7"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      6000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      100
    </Col>
  </Row>
  <Row
    className="h5"
    key="8"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      7000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      200
    </Col>
  </Row>
  <Row
    className="h5"
    key="9"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      8000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      300
    </Col>
  </Row>
  <Row
    className="h5"
    key="10"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      8700
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      310
    </Col>
  </Row>
  <Row
    className="h5"
    key="11"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      9900
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      100
    </Col>
  </Row>
  <Row
    className="h5"
    key="12"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      9000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      350
    </Col>
  </Row>
  <Row
    className="h5"
    key="13"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      10000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      420
    </Col>
  </Row>
  <Row
    className="h5"
    key="14"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      11500
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      440
    </Col>
  </Row>
</Fragment>
`;

exports[`<ListOfOrders /> render render orders for sell 1`] = `
<Fragment>
  <Row
    className="h5"
    key="0"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      7000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      40
    </Col>
  </Row>
  <Row
    className="h5"
    key="1"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      8000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      100
    </Col>
  </Row>
  <Row
    className="h5"
    key="2"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      5000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      100
    </Col>
  </Row>
  <Row
    className="h5"
    key="3"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      11000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      400
    </Col>
  </Row>
  <Row
    className="h5"
    key="4"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      22000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      602
    </Col>
  </Row>
  <Row
    className="h5"
    key="5"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      8000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      100
    </Col>
  </Row>
  <Row
    className="h5"
    key="6"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      9900
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      590
    </Col>
  </Row>
  <Row
    className="h5"
    key="7"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      10000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      600
    </Col>
  </Row>
  <Row
    className="h5"
    key="8"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      9900
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      400
    </Col>
  </Row>
  <Row
    className="h5"
    key="9"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      11400
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      630
    </Col>
  </Row>
  <Row
    className="h5"
    key="10"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      12000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      670
    </Col>
  </Row>
  <Row
    className="h5"
    key="11"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      10000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      630
    </Col>
  </Row>
  <Row
    className="h5"
    key="12"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      50000
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      670
    </Col>
  </Row>
  <Row
    className="h5"
    key="13"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      22400
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      550
    </Col>
  </Row>
  <Row
    className="h5"
    key="14"
    tag="div"
  >
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      $
      23500
    </Col>
    <Col
      lg="2"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      x
    </Col>
    <Col
      lg="5"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      700
    </Col>
  </Row>
</Fragment>
`;
